octoprint/assets/configuration/spec.yaml

```diff
@@ -0,0 +1,48 @@
+name: octoprint
+files:
+- name: octoprint.yaml
+  options:
+  - template: init_config
+    options:
+    - template: init_config/default
+  - template: instances
+    options:
+    - name: url
+      required: true
+      description: The URL to check.
+      value:
+        type: string
+        example: http://octoprint.local
+    - name: octo_api_key
+      required: true
+      description: API key to your OctoPrint instance
+      value:
+        type: string
+        example: ABC123
+    - name: search_string
+      required: true
+      description: The string to search for.
+      value:
+        type: string
+        example: OctoPrint
+    - name: flag_follow_redirects
+      required: false
+      description: Follow 301 redirects.
+      value:
+        type: boolean
+        example: false
+    - template: instances/default
+  - template: logs
+    example:
+    - type: file
+      path: "/home/pi/.octoprint/logs"
+      service: "octoprint"
+      source: "octoprint"
+    - type: file
+      path: "/var/log/webcamd.log"
+      service: "octoprint-webcam"
+      source: "octoprint"
+    - type: file
+      path: "/var/log/haproxy.log"
+      service: "octoprint-haproxy"
+      source: "octoprint"
```

Comment:
Were you able to test this? AFAIK we currently don't support `extras` integration logs.